api/api.go

```diff
@@ -764,15 +761,12 @@ func (api *Server) readState(ctx context.Context, in *iotexapi.ReadStateRequest)
 	if err != nil {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
-	data, err := p.ReadState(ctx, ws, in.MethodName, in.Arguments...)
+	data, err := p.ReadState(ctx, ws, methodName, arguments...)
 	// TODO: need to distinguish user error and system error
 	if err != nil {
 		return nil, err
 	}
-	out := iotexapi.ReadStateResponse{
-		Data: data,
-	}
-	return &out, nil
+	return data, nil
```

Comment:
`return p.ReadState(ctx, ws, methodName, arguments...)`

Reply:
fixed in latest commit